src/java.base/share/classes/java/util/FormatItem.java

```diff
@@ -0,0 +1,570 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.util;
+
+import java.io.IOException;
+import java.lang.invoke.*;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+import java.text.DecimalFormatSymbols;
+import java.util.Formatter.FormatSpecifier;
+
+import jdk.internal.access.JavaLangAccess;
+import jdk.internal.access.SharedSecrets;
+import jdk.internal.javac.PreviewFeature;
+import jdk.internal.util.Digits;
+import jdk.internal.util.Digits.*;
+import jdk.internal.util.FormatConcatItem;
+
+import static java.lang.invoke.MethodType.methodType;
+
+/**
+ * A specialized objects used by FormatterBuilder that knows how to insert
+ * themselves into a concatenation performed by StringConcatFactory.
+ *
+ * @since 19
+ */
+@PreviewFeature(feature=PreviewFeature.Feature.STRING_TEMPLATES)
+class FormatItem {
+    private static final JavaLangAccess JLA = SharedSecrets.getJavaLangAccess();
+
+    private static final MethodHandle CHAR_MIX =
+            JLA.stringConcatHelper("mix",
+                    MethodType.methodType(long.class, long.class,char.class));
+
+    private static final MethodHandle STRING_PREPEND =
+            JLA.stringConcatHelper("prepend",
+                    MethodType.methodType(long.class, long.class, byte[].class,
+                            String.class, String.class));
+
+    private static final MethodHandle SELECT_GETCHAR_MH =
+            JLA.stringConcatHelper("selectGetChar",
+                    MethodType.methodType(MethodHandle.class, long.class));
+
+    private static final MethodHandle SELECT_PUTCHAR_MH =
+            JLA.stringConcatHelper("selectPutChar",
+                    MethodType.methodType(MethodHandle.class, long.class));
+
+    private static final long charMix(long lengthCoder, char value) {
+        try {
+            return (long)CHAR_MIX.invokeExact(lengthCoder, value);
+        } catch (RuntimeException ex) {
```

Comment:
This should also rethrow `Error` instances:
```suggestion
        } catch (Error | RuntimeException ex) {
```

Reply:
Changing